workflow-manager/src/workflow_update.rs

```diff
@@ -0,0 +1,284 @@
+                        funds: vec![]
+                    }.into());
+
+                    // Modify saved config with the new modified authorizations
+                    let auth = config
+                        .authorizations
+                        .iter_mut()
+                        .find(|a| a.label == label)
+                        .context(format!("Failed to find authorization {}", label))?;
+
+                    if let Some(not_before) = not_before {
+                        auth.not_before = not_before;
+                    }
+
+                    auth.priority = priority;
+                    auth.max_concurrent_executions = max_concurrent_executions;
+                }
+                AuthorizationInfoUpdate::Disable(label) => {
+                    verify_authorization_exists(&config.authorizations, label.clone())?;
+
+                    // Create instruction for disabling authorization
+                    instructions.push_back(WasmMsg::Execute {
+                        contract_addr: config.authorization_data.authorization_addr.clone(),
+                        msg: to_json_binary(&valence_authorization_utils::msg::ExecuteMsg::PermissionedAction(
+                            valence_authorization_utils::msg::PermissionedMsg::DisableAuthorization { label }
+                        )).unwrap(),
+                        funds: vec![]
+                    }.into());
+                }
+                AuthorizationInfoUpdate::Enable(label) => {
+                    verify_authorization_exists(&config.authorizations, label.clone())?;
+
+                    // Create instruction for enabling authorization
+                    instructions.push_back(WasmMsg::Execute {
+                        contract_addr: config.authorization_data.authorization_addr.clone(),
+                        msg: to_json_binary(&valence_authorization_utils::msg::ExecuteMsg::PermissionedAction(
+                            valence_authorization_utils::msg::PermissionedMsg::EnableAuthorization { label }
+                        )).unwrap(),
+                        funds: vec![]
+                    }.into());
+                }
+            }
+        }
+
+        // Add all new authorizations
+        instructions.push_front(
+            WasmMsg::Execute {
+                contract_addr: config.authorization_data.authorization_addr.clone(),
+                msg: to_json_binary(
+                    &valence_authorization_utils::msg::ExecuteMsg::PermissionedAction(
+                        valence_authorization_utils::msg::PermissionedMsg::CreateAuthorizations {
+                            authorizations: new_authorizations,
+                        },
+                    ),
+                )
+                .unwrap(),
+                funds: vec![],
+            }
+            .into(),
+        );
+
+        // Save the new config to the registry
+        neutron_connector.update_workflow_config(config).await?;
+
```

Comment:
General comment: there's a lot happening in this function, and those are potentially sensitive operations (as in, breaking) on all aspects of a workflow/program config. I think it would be good (for future maintenance's sake) to organize the logic in a clear, staged structure .. as the current logic is mixing updating the authorizations, creating instructions, .. it's easy to get lost in what's happening.

Reply:
It might not be clear what update does.

Update should not break the program, what it does is it allows updating library configs and authorizations, thats it, the manager cannot execute anything, and this is what instructions are, the update function output the messages that needs to be executed.

so there are 3 things happening here:
1. We build the messages to add/modify/disable/enable **authorizations**
2. We create the message to update the **library config**
3. Only the processor can update library configs because its the owner of the library contract, so we need to build the authorization that allows this operation.

After the update method ran, it output what messages needs to be executed by the owner in order to perform all those changes.

Reply:
I understand .. but as you mentioned, there are 3 things happening .. and the logic for those 3 things is interweaved / inter-related. I just suggest that making that code a bit clearer would help with future maintenance.

Reply:
Added comments to explain a little more whats happeneing